### PR TITLE
Add note on emscripten_set_main_loop with Wasm EH

### PR DESCRIPTION
**Pull request by aheejin**
When `simulate_infinite_loop` is true, `emscripten_set_main_loop` does
not work with [the new Wasm EH](https://emscripten.org/docs/porting/exceptions.html#webassembly-exception-handling-proposal)
because throwing a JS exception will cause destructors in the Wasm stack
frames to run. This can be fixed in future by either by fixing the LLVM
compilation by adding an additional `catch` to every `catch_all` so that
we can rethrow JS exceptions of this kind, or possibly using [the new
stack switching proposal](https://github.com/WebAssembly/stack-switching).

This PR adds a note that this features currently doesn't work with Wasm
EH.

### CODE REVIEW

**Comment by sbc100** [comment]
This is rather unfortunate...   given that it doesn't work can we assert at runtime if this flag is passed?  Or at least write a warning the console?  I guess it only really matters if you have C++ object in your stack when you call it so many users (e.g SDL games who call this from `main`) can still work fine?

File: site/source/docs/api_reference/emscripten.h.rst [hunk 1]
```diff
@@ -361,6 +361,8 @@ Functions
 
   .. note:: Calling this function overrides the effect of any previous calls to :c:func:`emscripten_set_main_loop_timing` in the calling thread by applying the timing mode specified by the parameter ``fps``. To specify a different timing mode for the current thread, call the function :c:func:`emscripten_set_main_loop_timing` after setting up the main loop.
 
+  .. note:: Currently, ``simulate_infinite_loop`` == true does not work yet with `the new Wasm exception handling <https://emscripten.org/docs/porting/exceptions.html#webassembly-exception-handling-proposal>`_; your program may crash with memory errors.
```
Comment (sbc100) [inline site/source/docs/api_reference/emscripten.h.rst:364]: Perhaps mention "does not work in C++ projects that have objects with destructors on the stack at the time of the call"?

**Comment by aheejin** [comment]
> This is rather unfortunate... given that it doesn't work can we assert at runtime if this flag is passed? Or at least write a warning the console? I guess it only really matters if you have C++ object in your stack when you call it so many users (e.g SDL games who call this from `main`) can still work fine?

I think you're right; if you don't have any destructors to run this will work fine. So if you only call this from the main without any stack-allocated objects this should be fine. So I'm not sure if we should assert... maybe a warning is enough? (Then where should we warn? Inside `emscripten_set_main_loop`?

File: site/source/docs/api_reference/emscripten.h.rst (hunk 1 above)
Comment (aheejin) [reply site/source/docs/api_reference/emscripten.h.rst:364]: Done

**Review by kripken — APPROVED** [review]

**Pull request merged**